lib/poller.js

```diff
@@ -150,37 +189,66 @@ class Poller {
   }
 
   /**
-   * Checks if secret exists, if not trigger a poll
+   * Checks status of external secret and determines time to next poll
+   * If current observed generation is older than ES generation it will poll right away
+   * otherwise check when it was last polled and set timeout for next poll
    */
-  async _checkForSecret () {
-    const kubeNamespace = this._kubeClient.api.v1.namespaces(this._namespace)
-
+  async _checkExternalSecretStatus () {
     try {
-      await kubeNamespace.secrets(this._secretDescriptor.name).get()
-    } catch (err) {
-      if (err.statusCode === 404) {
-        this._logger.info(`Secret ${this._secretDescriptor.name} does not exist, polling right away`)
-        this._poll()
+      const {
+        body: {
+          status: {
+            lastSync = null,
+            observedGeneration = 0
+          } = {}
+        } = {}
+      } = await this._status.get()
+
+      const currentGeneration = this._externalSecret.metadata.generation
+
+      if (observedGeneration < currentGeneration) {
+        return this._setNextPoll(0)
+      }
+
+      const lastPollNbr = Date.parse(lastSync) || 0
+      const now = Date.now()
+
+      if (lastPollNbr > now) {
```

Comment:
In what cases would you expect this to be true?

Reply:
Well never TBH, unless someone messes up their time zones or so :D I just ran into it when messing with the tests (intentionally setting bad dates). Could remove it